File: visual-model-app/src/main/java/org/visual/model/app/module/UIModule.java
package org.visual.model.app.module;

import com.google.inject.AbstractModule;
import javafx.stage.Stage;
import org.visual.model.app.provider.RootStageProvider;

public class UIModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(Stage.class).toProvider(RootStageProvider.class);
    }
}

File: visual-model-app/src/main/kotlin/org/visual/model/app/module/UIModule.kt
package org.visual.model.app.module

import com.google.inject.AbstractModule
import javafx.stage.Stage
import javafx.stage.StageStyle
import org.visual.model.shared.KSlf4j
import org.visual.model.shared.KSlf4j.Companion.log

@KSlf4j
class UIModule : AbstractModule() {
    private val rootStage = Stage().apply {
        centerOnScreen()
        initStyle(StageStyle.TRANSPARENT)
        isResizable = true
    }

    override fun configure() {
        log.info("ui module start")
        bind(Stage::class.java).toInstance(rootStage)
    }
}
